### PR TITLE
HADOOP-18627 Stronger wording in 'secure mode' intro

### PR DESCRIPTION
**Pull request by raboof**
### Description of PR

Make it more clear that when exposing Hadoop to untrusted users, 'secure mode' is not optional.

### How was this patch tested?

n/a

### CODE REVIEW

**Comment by hadoop-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 50s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  1s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  1s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  1s |  |  detect-secrets was not available.  |
| +0 :ok: |  markdownlint  |   0m  1s |  |  markdownlint was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
|||| _ trunk Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  52m  4s |  |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   1m 30s |  |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  79m 26s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   1m  8s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  mvnsite  |   1m 21s |  |  the patch passed  |
| +1 :green_heart: |  shadedclient  |  27m  7s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 35s |  |  The patch does not generate ASF License warnings.  |
|  |   | 110m 54s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.42 ServerAPI=1.42 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5406/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/5406 |
| Optional Tests | dupname asflicense mvnsite codespell detsecrets markdownlint |
| uname | Linux cec8197a2d0f 4.15.0-197-generic #208-Ubuntu SMP Tue Nov 1 17:23:37 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / 459719ea554f4f181d9feb86aacc0ab23fce2dbc |
| Max. process+thread count | 534 (vs. ulimit of 5500) |
| modules | C: hadoop-common-project/hadoop-common U: hadoop-common-project/hadoop-common |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5406/1/console |
| versions | git=2.25.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

File: hadoop-common-project/hadoop-common/src/site/markdown/SecureMode.md [hunk 1]
```diff
@@ -20,7 +20,9 @@ Hadoop in Secure Mode
 Introduction
 ------------
 
-This document describes how to configure authentication for Hadoop in secure mode. When Hadoop is configured to run in secure mode, each Hadoop service and each user must be authenticated by Kerberos.
+In its default configuration, we expect you to make sure attackers don't have access to your Hadoop deployment by restricting all network access. If you want to expose Hadoop to untrusted users, you will have to configure authentication for Hadoop in secure mode as described in this document.
```
Comment (steveloughran) [inline hadoop-common-project/hadoop-common/src/site/markdown/SecureMode.md]: I don't think people cloud deployments realise they've exposed their hadoop services to the world, so we need to be a bit more explicit here.

**Review by steveloughran — COMMENTED** [review]
I like the extra text, but we should maybe make it even clearer about deployments and the default accessibility "unless you do something your cluster is exposed to everyone else on the same network. in cloud that is a lot"

**Review by steveloughran — APPROVED** [review]
+1 after my tuning; merging everywhere

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m  0s |  |  Docker mode activated.  |
| -1 :x: |  patch  |   0m 21s |  |  https://github.com/apache/hadoop/pull/5406 does not apply to trunk. Rebase required? Wrong Branch? See https://cwiki.apache.org/confluence/display/HADOOP/How+To+Contribute for help.  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| GITHUB PR | https://github.com/apache/hadoop/pull/5406 |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5406/2/console |
| versions | git=2.17.1 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m  0s |  |  Docker mode activated.  |
| -1 :x: |  patch  |   0m 21s |  |  https://github.com/apache/hadoop/pull/5406 does not apply to trunk. Rebase required? Wrong Branch? See https://cwiki.apache.org/confluence/display/HADOOP/How+To+Contribute for help.  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| GITHUB PR | https://github.com/apache/hadoop/pull/5406 |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-5406/3/console |
| versions | git=2.17.1 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

**Pull request merged**